datalad/customremotes/datalad.py

```diff
@@ -52,12 +52,10 @@ def transfer_retrieve(self, key, file):
                 return
             except Exception as exc:
                 ce = CapturedException(exc)
-                cause = getattr(exc, '__cause__', None)
-                debug_msg = f"Failed to download {url} for key {key}: {ce}"
-                if cause:
-                    debug_msg += f' [{cause}]'
+                debug_msg = f"Failed to download {url} for key {key}: " \
+                            f"{ce.format_with_cause()}"
                 self.message(debug_msg)
-                error_causes.append(cause)
+                error_causes.append(ce.tb.__cause__)
```

Comment:
I don't see complaints from codecov which means that this code block is probably tests covered -- could you add checking for expected cause to appear somewhere where expected in the test?

Reply:
I don't see where this is covered. `test_datalad.py` doesn't appear to test retrieval at all.